include/openPMD/IO/ADIOS/ADIOS2Auxiliary.hpp

```diff
@@ -34,8 +36,28 @@
 #include <utility>
 #include <vector>
 
+/*
+ * ADIOS2 v2.8 brings mode::ReadRandomAccess
+ */
+#define HAS_ADIOS_2_8 (ADIOS2_VERSION_MAJOR * 100 + ADIOS2_VERSION_MINOR >= 208)
+/*
+ * ADIOS2 v2.9 brings modifiable attributes (technically already in v2.8, but
+ * there are too many bugs, so we only support it beginning with v2.9).
+ * Group table feature requires ADIOS2 v2.9.
+ */
+#define HAS_ADIOS_2_9 (ADIOS2_VERSION_MAJOR * 100 + ADIOS2_VERSION_MINOR >= 209)
```

Comment:
I wonder if these should be runtime functions instead?

Anyway, the way you use them here macros make sense.
One thing to consider: this is a header file. You might want to undefine the two macros at the end of it, especially with the generic name.

Since you use the macros around a few locations (e.g., also in tests), maybe provide them in a helper header file?